src/store/pool/hooks.ts

```diff
@@ -355,3 +357,63 @@ export function useAPYs() {
 
   return apys;
 }
+
+export enum PairState {
+  LOADING,
+  NOT_EXISTS,
+  EXISTS,
+  INVALID,
+}
+
+export function useV2Pairs(currencies: [Currency | undefined, Currency | undefined][]): [PairState, Pair | null][] {
+  const [reserves, setReserves] = useState<({ reserve0: string; reserve1: string } | number | undefined)[]>([]);
+
+  const tokens = useMemo(() => {
+    setReserves(Array(currencies.length).fill(PairState.LOADING));
+    return currencies.map(([currencyA, currencyB]) => [currencyA?.wrapped, currencyB?.wrapped]);
+  }, [currencies]);
+
+  useEffect(() => {
+    const fetchReserves = async () => {
+      const result = await Promise.all(
+        tokens.map(async ([tokenA, tokenB]) => {
+          if (tokenA && tokenB && tokenA.chainId === tokenB.chainId && !tokenA.equals(tokenB)) {
+            const poolId = await bnJs.Dex.getPoolId(tokenA.address, tokenB.address);
+            const reserveA = await bnJs.Dex.getPoolTotal(poolId, tokenA.address);
```

Comment:
what about using `getPoolStats` API? It can reduce API calls.

Reply:
Ah, yes!